crates/miden-testing/tests/auth/multisig.rs

```diff
@@ -0,0 +1,400 @@
+    let tx_summary = SigningInputs::TransactionSummary(tx_summary);
+
+    let sig_1 = authenticators[0].get_signature(public_keys[0].into(), &tx_summary)?;
+    let sig_2 = authenticators[1].get_signature(public_keys[1].into(), &tx_summary)?;
+
+    // Populate advice map with signatures
+    let mut advice_map = AdviceMap::default();
+    advice_map.insert(Hasher::merge(&[public_keys[0].into(), msg]), sig_1);
+    advice_map.insert(Hasher::merge(&[public_keys[1].into(), msg]), sig_2);
+
+    // Execute transaction with signatures - should succeed
+    let tx_context_execute = mock_chain
+        .build_tx_context(multisig_account.id(), &[], &[])?
+        .tx_script(tx_script)
+        .extend_expected_output_notes(vec![OutputNote::Full(output_note)])
+        .extend_advice_map(advice_map)
+        .auth_args(salt)
+        .build()?;
+    tx_context_execute.execute().expect("Transaction should succeed");
+
+    Ok(())
+}
+
+#[test]
+fn test_multisig_4_owners_threshold_2() -> anyhow::Result<()> {
+    // Test 4 owners with threshold 2 - only 2 signatures should be needed
+
+    // Setup keys and authenticators (4 approvers, but only 2 signers)
+    let (_secret_keys, public_keys, authenticators) =
+        setup_keys_and_authenticators(4, 2, [1u8; 32])?;
+
+    // Create multisig account with 4 approvers but threshold of 2
+    let multisig_account = create_multisig_account(2, &public_keys, [1; 32], 10)?;
+
+    let mut mock_chain = MockChainBuilder::with_accounts([multisig_account.clone()])
+        .unwrap()
+        .build()
+        .unwrap();
+
+    mock_chain.prove_next_block()?;
+
+    // Create output note for the transaction
+    let output_note = create_p2id_note(
+        ACCOUNT_ID_REGULAR_PUBLIC_ACCOUNT_IMMUTABLE_CODE.try_into().unwrap(),
+        ACCOUNT_ID_REGULAR_PUBLIC_ACCOUNT_UPDATABLE_CODE.try_into().unwrap(),
+        vec![],
+        NoteType::Public,
+        Default::default(),
+        &mut RpoRandomCoin::new(Word::from([4u32; 4])),
+    )?;
+
+    let asset = FungibleAsset::mock(5);
+    let tx_script = TransactionScript::compile(
+        format!(
+            "
+            use.miden::tx
+            begin
+                push.{recipient}
+                push.{note_execution_hint}
+                push.{note_type}
+                push.0              # aux
+                push.{tag}
+                call.tx::create_note
+
+                push.{asset}
+                call.::miden::contracts::wallets::basic::move_asset_to_note
+                dropw dropw dropw dropw
+            end
+            ",
+            recipient = word_to_masm_push_string(&output_note.recipient().digest()),
+            note_type = NoteType::Public as u8,
+            tag = Felt::from(output_note.metadata().tag()),
+            asset = word_to_masm_push_string(&asset.into()),
+            note_execution_hint = Felt::from(output_note.metadata().execution_hint()),
+        ),
+        TransactionKernel::testing_assembler(),
+    )?;
```

Comment:
Since the issue https://github.com/0xMiden/miden-base/issues/1646 turned out NOT to be the culprit, I think we can safely reduce the code size of the tests and not have a tx script at all